engine/src/cmdss.cpp

```diff
@@ -448,16 +448,22 @@ Parse_stat MCGo::parse(MCScriptPoint &sp)
 MCStack *MCGo::findstack(MCExecContext &ctxt, MCStringRef p_value, Chunk_term etype, MCCard *&cptr)
 {
     MCStack *sptr = nil;
+    sptr = MCInterfaceTryToEvalStackFromString(p_value);
+    if (sptr != nil)
+        return sptr;
+
     if (MCInterfaceStringCouldBeStack(p_value))
     {
-        sptr = MCInterfaceTryToEvalStackFromString(p_value);
-        if (sptr == nil)
-        {
+        //sptr = MCInterfaceTryToEvalStackFromString(p_value);
+        //if (sptr == nil)
+        //{
             if (MCresult->isclear())
                 ctxt . SetTheResultToCString("can't build stack from string");
-        }
+        //}
         return sptr;
     }
+    else
+        MCresult->clear();
     
 	if (etype == CT_STACK)
         return NULL;
```

Comment:
Why is nil returned for some cases and NULL in others?  There are 2 calls in MCGo::exec_ctxt to findstack.  The first uses (sptr==nil) and the second uses (sptr == NULL) to test for success.  My understanding is that the difference is the type (nullptr vice int).  In foundation.h there is a define for nil (C++ compiler) to nullptr.